feat: add cefWindowGotFocus signal to notify if CefWindow got focus

philippun1 opened the pull request:
This MR adds a cefWindowGotFocus signal to QCefView and emits the signal if the CefWindow gets the focus.

This change allows to detect if the internal CefWindow got focus in both OSR mode on and off.

tishion commented:
Hi @philippun1, thanks for your contribution. Before we can move on to process this PR, I have some questions:
1. what problem can this PR solve?
2. if you just want to get the focusIn event of CEF native window, is this method enough for it QCefViewPrivate::onCefWindowGotFocus()
https://github.com/CefView/QCefView/blob/3c0bfe0a693c2d14b4a442958f497a2f2a2b4af3/src/details/QCefViewPrivate.cpp#L500-L514

My understand is currently in NCW mode, when CEF native window got focus, we cannot get this event from QCefView instance, right?

philippun1 commented:
@tishion thanks for the review.
1. I have a use case where the QCefView with OSR disabled is part of a bigger Qt GUI and whenever it gets focus I want to make updates to the other GUI elements. With OSR disabled the QCefView does not get the regular Qt events like focusIn. So this signal would allow me to get notified about it.
2. This method is enough and I think an even better solution. I updated the code accordingly.

I renamed my signal to `cefWindowGotFocus` to reflect that it now forwards the `onCefWindowGotFocus` instead of a Qt `focusIn` event and the signal gets emitted directly in `onCefWindowGotFocus`. This works with both OSR mode off and on.

tishion commented:
@philippun1 
I totally understood your requirement now.
I think missing the FocusIn/Out events in NCW mode is a real problem. Instead of adding a new signal, I prefer to implement it by triggering the focusIn/Out event. I will draft a change to achieve this.  Let's keep this PR open, in case that I am not able to finish it. Is this good to you?

philippun1 commented:
@tishion Sounds good, thank you.

tishion commented:
Resolved with #495.
I am closing this PR. 
Thank you.
